backend/storage-service/src/jvmMain/kotlin/file/services/acl/AclService.kt

```diff
@@ -287,6 +296,110 @@ class AclService(
+        data class SynchronizedFolderWithNewAccess(
+            val id: String,
+            val user: String,
+            val path: String,
+            val oldType: SynchronizationType,
+            val newType: SynchronizationType?
+        )
+
+        db.withSession { session ->
+            val syncedChildren = session.sendPreparedStatement(
+                {
+                    setParameter("path", path)
+                },
+                """
+                    select id, user_id, path, access_type from storage.synchronized_folders
+                    where path like :path || '%'
+                """
+            ).rows
+
+            val changes: List<SynchronizedFolderWithNewAccess> = syncedChildren.map {
+                val newType = if (hasPermission(
+                        it.getField(SynchronizedFoldersTable.path),
+                        it.getField(SynchronizedFoldersTable.user),
+                        AccessRight.WRITE
+                    )
+                ) {
+                    SynchronizationType.SEND_RECEIVE
+                } else if (hasPermission(
+                        it.getField(SynchronizedFoldersTable.path),
+                        it.getField(SynchronizedFoldersTable.user),
+                        AccessRight.READ
+                    )
+                ) {
+                    SynchronizationType.SEND_ONLY
+                } else {
+                    null
+                }
+
+                SynchronizedFolderWithNewAccess(
+                    it.getField(SynchronizedFoldersTable.id),
+                    it.getField(SynchronizedFoldersTable.user),
+                    it.getField(SynchronizedFoldersTable.path),
+                    SynchronizationType.valueOf(it.getField(SynchronizedFoldersTable.accessType)),
+                    newType
+                )
+            }
+
+            val toUpdate = changes.filter { it.newType != null }
+            val toDelete = changes.filter { it.newType == null }
+
+            if (toUpdate.isNotEmpty()) {
+                session.sendPreparedStatement(
+                    {
+                        setParameter("ids", toUpdate.map { it.id })
+                        setParameter("users", toUpdate.map { it.user })
+                        setParameter("access", toUpdate.map { it.newType })
+                    },
+                    """
+                        update storage.synchronized_folders f
+                        set access_type = new.access
+                        from (
+                            select
+                                unnest(:ids::text[]) as id,
+                                unnest(:users::text[]) as user,
+                                unnest(:access::text[]) as access
+                        ) as new
+                        where f.id = new.id and f.user_id = new.user
+                    """
+                )
+
+                Mounts.mount.call(
```

Comment:
Does the `mount` call correctly handling this potentially already being present?

Reply:
Mount will attempt to unmount any folders which already exist, before attempting to mount.

However, the call to mount in AclService shouldn't actually be there at all, since the list of folders to mount will always be empty. I have removed it.

Reason: If permissions change such that a user no longer have (at least) read permission, the folder will be removed completely from syncthing and unmounted. If the user regains permission to the directory, we cannot remount it since we have removed the association completely.

We can change this such that it doesn't delete the entry from the database (like it does now), but just from Syncthing + unmount. In that case we can remount the folder if the user regains permission to the folder.